src/redux/reducer.js

```diff
@@ -348,13 +356,18 @@ export const reducer = createReducer(initialState, {
     },
 });
 
-function synchCurrentTreeNode(state) {
+function synchCurrentTreeNode(state, parentNodeId = undefined) {
     const currentNode = state.networkModificationTreeModel?.treeElements.find(
         (entry) => entry?.id === state.currentTreeNode?.id
     );
-    // handle the case of currentTreeNode not in the TreeModel anymore.
-    if (currentNode === undefined) state.currentTreeNode = null;
-    else {
+    // handle the case of currentTreeNode is deleted
+    if (currentNode === undefined) {
+        const parentNode =
+            state.networkModificationTreeModel?.treeElements.find(
+                (treeElement) => treeElement?.id === parentNodeId
+            );
+        state.currentTreeNode = { ...parentNode };
+    } else {
         state.currentTreeNode = { ...currentNode };
```

Comment:
Add a comment that we need to overwrite state.currentTreeNode line 371 to consider label change for example.

Reply:
Ok, done.